migrationrunners/migrationrunners.go

```diff
@@ -26,25 +27,16 @@ func New(server servers.IServer) (*Runner, error) {
 
 // RunOutstanding runs all outstanding migrations
 func (r *Runner) RunOutstanding() error {
-	migrationRepo, err := migrationloaders.Load()
+	migrationRepo, err := r.getOutstandingMigrations()
 	if err != nil {
 		return err
 	}
 
-	appliedMigrationVersions, err := r.getAppliedMigrationVersions()
-	if err != nil {
-		return err
-	}
-
-	for _, version := range appliedMigrationVersions {
-		delete(migrationRepo, version.Version)
-	}
-
 	versions := migrationRepo.SortedVersions()
 
 	for _, version := range versions {
 		mgr := migrationmanagers.NewWithDependencies(migrationRepo[version], r.server, r.schema)
-		err := mgr.Run(migrationRepo)
+		err := mgr.Migrate()
```

Comment:
And you can't get more pure than calling Migrate on individual migrations from within the `testtrack migrate` command.